File: add.py
#sum of two numbers


a=int(input("Enter a:"))
b=int(input("Enter b:"))
c=a+b
print(c)



#division of two numbers


a=input('enter number a:')
b=input('enter number b:')
acopy=a.replace("."," ")
bcopy=b.replace("."," ")
if((a.isnumeric() or acopy.isnumeric()) and (b.isnumeric() or bcopy.isnumeric())):
	print(float(a)/float(b))
else:
	print("enter a numerical value")
	
	
	
	
#dot product of two vectors




l=[]
m=[]
s=0
n=int(input('length of vector'))
for i in range(n):
	a=int(input('enter number'))
	b=int(input('enter another number'))
	l.append(a)
	m.append(b)
print('list l=',l)
print('list m=',m)
for i in range(len(l)):
	s=s+l[i]*m[i]
print('dot product of two vectors=',s)
		
		
		
		
#find the determinant of the matrix
		
import numpy as np
n_array=np.array([[1,2,3],[4,7,6],[7,8,9]])
print('numpy matrix is:')
print(n_array)
det=np.linalg.det(n_array)
print('det of matrix:',int(det))
		
		
		
#addition of matrices


a=[[9,2,8],[4,5,6],[9,3,2]]
b=[[1,2,3],[4,5,6],[7,8,9]]
c=[[4,9,6],[2,4,0],[1,0,6]]
d=[[0,0,0],[0,0,0],[0,0,0]]
for i in range(len(a)):
	for j in range(len(a[0])):
		d[i][j]=a[i][j]+b[i][j]+c[i][j]
for row in d:
	print(row)

File: add.m
%addition of 2 numbers


a=input('Enter a:');
b=input('Enter b:');
c=a+b;
disp(c);



% division of two numbers


a=input('Enter a:','s');
b=input('Enter b:','s');
if all(isstrprop(a,'digit')) && all(isstrprop(b,'digit'))
  disp(a/b)
else
  disp('Enter a numrical value')
end


%dot product of two numbers


m=[];
k=[];
s=0;
n=input('Enter number of dimensions of vector:');
for i=1:n
  a=input('Enter co-efficient of vector a:');
  b=input('Enter co-efficient of vector b:');
  m(i)=a;
  k(i)=b; 
end
disp('vector1=');
disp(m);
disp('vector2=');
disp(k);
for i=1:length(m)
  s=s+m(i)*k(i);
end
fprintf('dot product=%d \n',s)


%determinant of matrix


matrix=[1,2,3;4,5,6;7,8,9]
determinant=det(matrix);
disp(determinant);



%addition of matrices



A=[1,2,3;4,5,6;7,8,9];
B=[4,5,6;2,4,5;1,4,3];
C=A+B;
disp('matrix A:');
disp(A);
disp('matrix b:');
disp(B);
disp('matrix C=(A+B):');
disp(C);